### PR TITLE
Dev #49 search free text

### PR DESCRIPTION
**Pull request by krios82**
* Add PGSearch
* Change the `pg_search_documents` table, make the `content` column a `tsvector` with `gin` index
* Change the `pg_search_documents` table, add `searchable_name`, `searchable_created_at` and `searchable_updated_at` helper fields for sorting
* Change the PgSearch `rebuild_pg_search_documents` function to calculate the `tsvector` and fill the helper fields
* Add the search results page
* Add pagination with Kaminari (easier to style than WillPaginate)
* Add sorting

### CODE REVIEW

File: spec/system/search_page_spec.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  it 'Will find concepts' do
+    visit '/categories'
+    fill_in('search', with: Concept.first.name)
+    click_button('Search')
+
+    expect(page).to have_field('search')
+    expect(page).to have_text("Results for '#{Concept.first.name}'")
+    expect(page).to have_text(Concept.first.category.name.upcase)
+    expect(page).to have_text(Concept.first.description)
+  end
```
Comment (spikeheap) [inline spec/system/search_page_spec.rb:33]: Could we add an additional test to ensure it finds data elements?

File: app/javascript/src/application.scss [hunk 2]
```diff
@@ -15,10 +17,24 @@ $govuk-assets-path: '../../../node_modules/govuk-frontend/assets/';
   background-color: govuk-colour("grey-4");
 }
 
+.govuk-\!-display-inline-block {
+  display: inline-block !important;
+}
+
+.flex {
+  flex: 1;
+}
```
Comment (spikeheap) [inline app/javascript/src/application.scss]: Is this used? I can't see it elsewhere in this PR. Flex is supported by IE11, but has a bunch of bugs we need to be careful of (https://caniuse.com/#search=flex)

Comment (krios82) [reply app/javascript/src/application.scss]: Leftover, removed

File: spec/system/search_page_spec.rb (hunk 1 above)
Comment (krios82) [reply spec/system/search_page_spec.rb:33]: Added

**Comment by spikeheap** [comment]
👍

**Pull request merged**